Code/Framework/AzCore/AzCore/Settings/SettingsRegistryImpl.cpp

```diff
@@ -722,8 +726,12 @@ namespace AZ
             mergeApproach = JsonMergeApproach::JsonMergePatch;
             break;
         default:
-            AZ_Assert(false, "Provided format for merging settings into the Setting Registry is unsupported.");
-            return false;
+            MergeSettingsResult mergeResult;
```

Comment:
This compiled? I usually get a compile error whenever I try to declare a variable inside a `case` expression unless it was wrapped with scoping curly braces

Reply:
A compile error only occurs when the variable is used before it has been declared in a switch statement.

A compile warning would occur if there was variable shadowing, but I was careful to avoid that.